### PR TITLE
[ISSUE #6665]Optimize the process of truncateInvalidMsgs()

### LINKED ISSUE #6665: Optimize the process of truncateInvalidMsg().
### Before Creating the Bug Report

- [x] I found a bug, not just asking a question, which should be created in [GitHub Discussions](https://github.com/apache/rocketmq/discussions).
- [x] I have searched the [GitHub Issues](https://github.com/apache/rocketmq/issues) and [GitHub Discussions](https://github.com/apache/rocketmq/discussions)  of this repository and believe that this is not a duplicate.
- [x] I have confirmed that this bug belongs to the current repository, not other repositories of RocketMQ.

### Describe the Bug

The confirmOffset is not reseted after truncating.

### Steps to Reproduce

3 Controllers, 3 Brokers.

The calculating process of truncateInvalidMsg has something wrong here:

image

The reputOffset was wrongly calculated, which leads to more and more messages were truncated.

### What Did You Expect to See?

No message lost.

### What Did You See Instead?

image

### Additional Context

The reputFromOffset is re-calculated here:
image

It uses confirmOffset:
image
image

The reputFromOffset is reseted, but the confirmOffset is not reseted:
image

### PR DESCRIPTION
**Pull request by GenerousMan**
### Which Issue(s) This PR Fixes

Fixes #6665

### Brief Description

1. Optimize the process of truncateInvalidMsgs()
2. Reset the confirmOffset while truncating.
3. Add more logs while expanding syncStateSet.

### How Did You Test This Change?

Using openchaos.

### CODE REVIEW

File: store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java [hunk 1]
```diff
@@ -710,6 +710,7 @@ public void truncateDirtyFiles(long offsetToTruncate) {
             this.reputMessageService = new ConcurrentReputMessageService();
         }
 
+        setConfirmOffset(offsetToTruncate);
```
Comment (RongtongJin) [inline store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java:713]: Before merging #6618, confirmOffset had two different values in AutoSwitchHAService and commitlog.

File: store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAService.java [hunk 2]
```diff
@@ -309,6 +310,8 @@ public void maybeExpandInSyncStateSet(final Long slaveBrokerId, final long slave
         if (slaveMaxOffset >= confirmOffset) {
             final EpochEntry currentLeaderEpoch = this.epochCache.lastEntry();
             if (slaveMaxOffset >= currentLeaderEpoch.getStartOffset()) {
+                LOGGER.info("The slave {} has caught up, slaveMaxOffset: {}, confirmOffset: {}, epoch: {}, startOffset: {}.",
```
Comment (RongtongJin) [inline store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAService.java:313]: How about renaming startOffset to leader epoch startOffset?

File: store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java (hunk 1 above)
Comment (GenerousMan) [reply store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java:713]: Yes, after #6618 is merged, we should use the DefaultMessageStore. 
I will help you review the #6618

File: store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAService.java (hunk 2 above)
Comment (GenerousMan) [reply store/src/main/java/org/apache/rocketmq/store/ha/autoswitch/AutoSwitchHAService.java:313]: Sure.

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**